### ISSUE TITLE
UnfinishedMockingSessionException when using MockitoTestNGListener instead of openMocks

### ISSUE THREAD

**Issue by violette** [mockito/mockito-testng#40]
Hi,

I'm using the `@Listeners(MockitoTestNGListener.class)` (mockito-testng 0.2.10) in a project to avoid all those open/close mocks.

While is it is running fine with some of use cases, I found a strange behavior:
Here a simple test class:
```java
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.entry;
import static org.mockito.Mockito.when;

import java.util.Map;

import org.assertj.core.data.MapEntry;
import org.mockito.Mock;
import org.mockito.testng.MockitoTestNGListener;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Listeners;
import org.testng.annotations.Test;

@Listeners(MockitoTestNGListener.class)
public class MockitoTestNgListenerTest1 {

    @Mock
    private Map<String, Object> map;

    private MapEntry<String, String> entry;

    @SuppressWarnings("rawtypes")
    @BeforeTest
    public void setUp() {
        entry = entry("Boo", "Minsc");
        when(map.get(0)).thenReturn(entry);
    }

    @Test
    public void shouldWork() {
        assertThat(map.get(0)).isEqualTo(entry);
    }
}
```
No big deal. It runs ok with TestNG.

Now duplicate this class and rename it. Now launch the 2 classes and the exception is thrown. All  unit tests fail.

```
org.mockito.exceptions.misusing.UnfinishedMockingSessionException: 
Unfinished mocking session detected.
Previous MockitoSession was not concluded with 'finishMocking()'.
For examples of correct usage see javadoc for MockitoSession class.

	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
	at org.testng.internal.invokers.InvokedMethodListenerInvoker.invokeListener(InvokedMethodListenerInvoker.java:56)
	at org.testng.internal.BaseInvoker.runInvokedMethodListeners(BaseInvoker.java:55)
	at org.testng.internal.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:351)
	at org.testng.internal.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:320)
	at org.testng.TestRunner.invokeTestConfigurations(TestRunner.java:617)
	at org.testng.TestRunner.beforeRun(TestRunner.java:607)
	at org.testng.TestRunner.run(TestRunner.java:578)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.runSuites(TestNG.java:1069)
	at org.testng.TestNG.run(TestNG.java:1037)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:73)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)

```
Note that those simple tests run fine with open/close mocks (or if the `BeforeTest` is changed to `BeforeMethod`).

Thanks.
------- Maven dependencies used:
JDK 11.0.6 / 11.0.8
Mockito-testng 0.2.10
TestNG version 7.3.0
Mockito 3.7.7

**Comment by violette** [comment]
In fact, this error occurs each time a `BeforeTest` is used, even if the `BeforeTest` contains code unrelated to the mocks.

**Comment by slawekjaranowski** [comment]
I confirm using `BeforeTest` in any place in project tests is the cause of incorrect `MockingSession` finishing.

**Comment by slawekjaranowski** [comment]
After analyzing I found that we can start `MockitoSession` only for `BeforeMethod` and of course before each test method.

Before rest of `BeforeXXX` annotation we can not start `MockitoSession` because it must be connected with specific test method.

**Comment by slawekjaranowski** [comment]
@violette fix is ready, but mock will not be initialized for `BeforeTest` method.
Why do you want use `BeforeTest`?

**Comment by violette** [comment]
Hi,
Thanks for the quick reply.
Long story short: we decided to use the `@Listeners` to close the mocks in very huge projects that was using the previous and now deprecated 'initMocks()'. So we have some `BeforeTest` used. Most of them were changed to `@BeforeMethod` to avoid that issue but in some cases, we cannot do it. (I don't have all the use cases in mind).
I think this is an important limitation and that should be documented.
Thanks,
